editors/vscode/src/quick_picks.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+import * as vscode from "vscode";
+import * as fs from "fs-extra";
+import simpleGit from "simple-git";
+import * as path from "node:path";
+
+export async function newProject(context: vscode.ExtensionContext) {
+    type Language = "Node (JavaScript/TypeScript)" | "C++" | "Rust";
+
+    const language = await vscode.window.showQuickPick(
+        ["Node (JavaScript/TypeScript)", "C++", "Rust"],
+        {
+            placeHolder: "What language do you want to use?",
+        },
+    ) as Language;
+
+    if (!language) {
+        vscode.window.showErrorMessage("Language selection is required.");
+        return;
+    }
+
+    let repoUrl: string | undefined;
+    switch (language) {
+        case "Node (JavaScript/TypeScript)":
+            repoUrl = "https://github.com/slint-ui/slint-nodejs-template";
+            break;
+        case "C++":
+            repoUrl = "https://github.com/slint-ui/slint-cpp-template";
+            break;
+        case "Rust":
+            repoUrl = "https://github.com/slint-ui/slint-rust-template";
+            break;
+        default:
+            vscode.window.showErrorMessage("Invalid language selection.");
+            return;
+    }
+
+    let workspacePath: string | undefined =
+        vscode.workspace.workspaceFolders?.[0]?.uri.fsPath;
+
+    if (!workspacePath) {
+        const folderUris = await vscode.window.showOpenDialog({
+            canSelectFolders: true,
+            canSelectMany: false,
+            openLabel: "Select Folder",
+        });
+
+        if (!folderUris || folderUris.length === 0) {
+            vscode.window.showErrorMessage(
+                "Please select a folder to place the project in.",
+            );
+            return;
+        }
+
+        workspacePath = folderUris[0].fsPath;
+    }
+
+    const projectName = await vscode.window.showInputBox({
+        prompt: "Enter the name of the new project",
+    });
+
+    if (!projectName) {
+        vscode.window.showErrorMessage("Project name is required.");
+        return;
+    }
+    
+    let authorName: string | undefined;
+    if (language === "Rust") {
+        authorName = await vscode.window.showInputBox({
+            prompt: "Enter the author name",
+        });
+
+        if (!authorName) {
+            vscode.window.showErrorMessage("Author name is required for Rust projects.");
+            return;
+        }
+    }
+
+    const projectPath = path.join(workspacePath, projectName);
+
+    try {
+        await fs.ensureDir(projectPath);
+
+        const git = simpleGit();
+        await git.clone(repoUrl, projectPath);
```

Comment:
What I do not like here is this:

You clone a git directory... so *every* new project will include the history of our template. That is not nice. Maybe delete the .git folder agian after the fact so that it is really just a set of files grabbed from somewhere and does not force users into git?

Reply:
Good point. Fixed